filament/src/fg/fg/RenderTargetResourceEntry.cpp

```diff
@@ -0,0 +1,206 @@
+            entry.descriptor.usage |= usages[i];
+
+            // update attachment sample count if not specified and usage permits it
+            if (!entry.descriptor.samples &&
+                none(entry.descriptor.usage & backend::TextureUsage::SAMPLEABLE)) {
+                entry.descriptor.samples = descriptor.samples;
+            }
+
+            attachments |= flags[i];
+
+            // figure out the min/max dimensions across all attachments
+            const size_t level = attachment.getLevel();
+            const uint32_t w = details::FTexture::valueForLevel(level, entry.descriptor.width);
+            const uint32_t h = details::FTexture::valueForLevel(level, entry.descriptor.height);
+            minWidth = std::min(minWidth, w);
+            maxWidth = std::max(maxWidth, w);
+            minHeight = std::min(minHeight, h);
+            maxHeight = std::max(maxHeight, h);
+        }
+    }
+
+    if (any(attachments)) {
+        if (minWidth == maxWidth && minHeight == maxHeight) {
+            // All attachments' size match, we're good to go.
+            width = minWidth;
+            height = minHeight;
+        } else {
+            // TODO: what should we do here? Is it a user-error?
+            width = maxWidth;
+            height = maxHeight;
+        }
+
+        if (resource.params.viewport.width == 0 && resource.params.viewport.height == 0) {
+            resource.params.viewport.width = width;
+            resource.params.viewport.height = height;
+        }
+        resource.params.flags.clear = descriptor.clearFlags;
+    }
+}
+
+void RenderTargetResourceEntry::update(FrameGraph& fg, PassNode const& pass) noexcept {
+    auto& resource = getResource();
+
+    // this update at every pass
+    if (any(attachments)) {
+        // overwrite discard flags with the per-rendertarget (per-pass) computed value
+        resource.params.flags.discardStart = TargetBufferFlags::NONE;
+        resource.params.flags.discardEnd   = TargetBufferFlags::NONE;
+
+        static constexpr TargetBufferFlags flags[] = {
+                TargetBufferFlags::COLOR,
+                TargetBufferFlags::DEPTH,
+                TargetBufferFlags::STENCIL };
+
+        auto& resourceNodes = fg.mResourceNodes;
+        for (size_t i = 0; i < descriptor.attachments.textures.size(); i++) {
+            FrameGraphHandle attachment = descriptor.attachments.textures[i];
+            if (attachment.isValid()) {
+                if (resourceNodes[attachment.index]->resource->discardStart) {
+                    resource.params.flags.discardStart |= flags[i];
+                }
+                if (resourceNodes[attachment.index]->resource->discardEnd) {
+                    resource.params.flags.discardEnd |= flags[i];
+                }
+            }
+        }
+
+        // clear implies discarding the content of the buffer
+        resource.params.flags.discardStart |= resource.params.flags.clear;
+
+// FIXME
```

Comment:
Does this need fixed now?

Reply:
This is actually unclear -- I don't think it worked before either.  I will get back to this when this becomes a problem.